Sources/MapboxNavigation/CarPlayNavigationViewController.swift

```diff
@@ -692,10 +684,21 @@ open class CarPlayNavigationViewController: UIViewController, BuildingHighlighti
         let nextStep = progress.currentLegProgress.stepIndex + 1
         
         navigationMapView?.addArrow(route: progress.route, legIndex: legIndex, stepIndex: nextStep)
-        navigationMapView?.show([progress.route], legIndex: legIndex)
+        updateRouteLine(routeProgress: progress, coordinate: navigationService.router.location?.coordinate)
         navigationMapView?.showWaypoints(on: progress.route, legIndex: legIndex)
     }
     
+    func updateRouteLine(routeProgress: RouteProgress, coordinate: CLLocationCoordinate2D?) {
+        navigationMapView?.show([routeProgress.route], legIndex: routeProgress.legIndex)
+        if routeLineTracksTraversal {
```

Comment:
It's a matter of style, but it's easier to read code when we're using early return, up to you though:
```suggestion
    if !routeLineTracksTraversal {
        return
    }
```